packages/neuron-ui/src/components/NervosDAO/hooks.ts

```diff
@@ -436,35 +437,79 @@ export const useUpdateWithdrawList = ({
   setWithdrawList: React.Dispatch<React.SetStateAction<Map<string, string | null>>>
 }) =>
   useEffect(() => {
-    Promise.all(
-      records.map(async ({ outPoint, depositOutPoint, blockHash }) => {
-        if (!tipBlockHash) {
-          return null
-        }
-        const withdrawBlockHash = depositOutPoint ? blockHash : tipBlockHash
-        const formattedDepositOutPoint = depositOutPoint
-          ? {
-              txHash: depositOutPoint.txHash,
-              index: `0x${BigInt(depositOutPoint.index).toString(16)}`,
-            }
-          : {
-              txHash: outPoint.txHash,
-              index: `0x${BigInt(outPoint.index).toString(16)}`,
-            }
-        return calculateDaoMaximumWithdraw(formattedDepositOutPoint, withdrawBlockHash).catch(() => null)
-      })
-    )
-      .then(res => {
-        const withdrawList = new Map()
-        if (tipBlockHash) {
-          records.forEach((record, idx) => {
-            const key = getRecordKey(record)
-            withdrawList.set(key, res[idx])
+    if (!tipBlockHash) {
+      setWithdrawList(new Map())
+      return
+    }
+    const depositOutPointHashes = records.map(v => v.depositOutPoint?.txHash ?? v.outPoint.txHash)
+    const txMap = new Map<CKBComponents.Hash, CKBComponents.TransactionWithStatus>()
+    ckbCore.rpc
+      .createBatchRequest<'getTransaction', string[], CKBComponents.TransactionWithStatus[]>(
+        depositOutPointHashes.map(v => ['getTransaction', v])
+      )
+      .exec()
+      .then(txs => {
+        const committedTx = txs.filter(v => v.txStatus.status === 'committed')
```

Comment:
Transactions should not be filtered because `txMap.set(depositOutPointHashes[idx], tx)` is related to the order, the order would be incorrect if some txs are filtered out.

Or maybe `txMap.set(tx.hash, tx)` is enough here. If so, the `if(tx.txStatus.status === 'committed')` is redundant because `committedTx.forEach` has been filtered before.

Reply:
I refer to this [function](https://github.com/nervosnetwork/ckb-sdk-js/blob/674dfa575571a835911410b1d6a12f8cf5999170/packages/ckb-sdk-core/src/index.ts#L423). If the transaction is not committed, it will throw an exception and we will set it null.
And of course `txMap.set(tx.hash, tx)` is enough here. I will fix it.